From e12f54edfa3d062f2f0fd25df55468024d87eebe Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@dereferenced.net>
Date: Wed, 2 Sep 2015 13:48:08 -0600
Subject: [PATCH] fix visibility of bundled zlib symbols with mingw

When QT_VISIBILITY_AVAILABLE is defined, the zlib header would use
visibility("default") instead of Q_DECL_EXPORT, but Windows needs
dllexport for these symbols to be available to QtGui/QtNetwork.

In practice, Q_CORE_EXPORT always has the correct behavior for the
zlib symbols.
---
 src/3rdparty/zlib/zlib.h | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/src/3rdparty/zlib/zlib.h b/src/3rdparty/zlib/zlib.h
index 5229707..d0908ab 100644
--- a/src/3rdparty/zlib/zlib.h
+++ b/src/3rdparty/zlib/zlib.h
@@ -34,13 +34,8 @@
 #include "zconf.h"
 
 #include <qglobal.h>
-#if defined(QT_VISIBILITY_AVAILABLE)
-# undef ZEXTERN
-# define ZEXTERN __attribute__((visibility("default")))
-#else
-# undef ZEXTERN
-# define ZEXTERN Q_DECL_EXPORT
-#endif
+#undef ZEXTERN
+#define ZEXTERN Q_CORE_EXPORT
 
 #ifdef __cplusplus
 extern "C" {
-- 
1.9.5.github.0

